### PR TITLE
8262885: Shenandoah: FullGC prologue does not need to save/restore heap has_forwarded_object flag

### PR DESCRIPTION
**Pull request by zhengyu123**
Another trivial cleanup. FullGC prologue code does not change heap's has_forwarded_objects flag, there is no point to save/restore it.

Test:
- [x] hotspot_gc_shenandoah

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8262885](https://bugs.openjdk.java.net/browse/JDK-8262885): Shenandoah: FullGC prologue does not need to save/restore heap has_forwarded_object flag

### Reviewers
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**) ⚠️ Review applies to e22fe9df6d1975543c91053252a4343806ab5dde

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2797/head:pull/2797`
`$ git checkout pull/2797`

### CODE REVIEW

**Comment by shipilev** [comment]
Yeah, but this code protects from some downstream code we are calling from the prologue to accidentally drop the flag. At least it should be asserted that `has_forwarded_objects` has the same value at the end of the prologue as it was at the beginning. Or better just leave this for safety?

File: src/hotspot/share/gc/shenandoah/shenandoahFullGC.cpp [hunk 1]
```diff
@@ -182,6 +185,8 @@ void ShenandoahFullGC::do_it(GCCause::Cause gc_cause) {
     // The rest of prologue:
     BiasedLocking::preserve_marks();
     _preserved_marks->init(heap->workers()->active_workers());
+    
+    assert(heap->has_forwarded_objects() == has_forwarded_objects, "Not expect to change");
```
Comment (shipilev) [inline src/hotspot/share/gc/shenandoah/shenandoahFullGC.cpp:189]: "This should not change"?

**Review by shipilev — APPROVED** [review]
Good!

**Comment by zhengyu123** [comment]
> Yeah, but this code protects from some downstream code we are calling from the prologue to accidentally drop the flag. At least it should be asserted that `has_forwarded_objects` has the same value at the end of the prologue as it was at the beginning. Or better just leave this for safety?

Okay, I added assert. The restore code led me to search where it got overwritten, a bit confusing.

**Pull request closed without merging**